server/container/guice/guice-common/src/main/java/org/apache/james/utils/InMemoryMailRepositoryStore.java

```diff
@@ -100,6 +101,15 @@ public void init() throws Exception {
         return Optional.empty();
     }
 
+    @Override
+    public Stream<MailRepository> getByPath(MailRepositoryPath path) {
+        return destinationToRepositoryAssociations
+                .keySet()
+                .stream()
+                .filter((MailRepositoryUrl key) -> key.asString().substring(key.getProtocol().getValue().length() + 3).equals(path.asString()))
```

Comment:
We should have a method of `Protocol` doing that substring operation. We can call it : `getProtocolIdentifier`

Reply:
other implementation(`MailRepositoryStoreBeanFactory`, `MockMailRepositoryStore`) i see `+ 2` not `+ 3`. Are they correct?